src/super_gradients/training/transforms/keypoints/keypoints_compose.py

```diff
@@ -0,0 +1,132 @@
+        if KeypointsMixup in self.transforms or KeypointsMosaic in self.transforms:
+            raise RuntimeError("KeypointsMixup & KeypointsMosaic augmentations are not supported in old-style transforms API")
+
+        for t in self.transforms:
+            image, mask, joints, areas, bboxes = t(image, mask, joints, areas, bboxes)
+
+        return image, mask, joints, areas, bboxes
+
+    def apply_to_sample(self, sample: PoseEstimationSample) -> PoseEstimationSample:
+        """
+        Applies the series of transformations to the input sample.
+        The function may modify the input sample inplace, so input sample should not be used after the call.
+
+        :param sample: Input sample
+        :return:       Transformed sample.
+        """
+        sample = sample.sanitize_sample()
+        sample = self._apply_transforms(
+            sample, transforms=self.transforms, load_sample_fn=self.load_sample_fn, min_bbox_area=self.min_bbox_area, min_visible_joints=self.min_visible_joints
+        )
+        return sample
+
+    @classmethod
+    def _apply_transforms(
+        cls, sample: PoseEstimationSample, transforms: List[AbstractKeypointTransform], load_sample_fn, min_bbox_area, min_visible_joints
+    ) -> PoseEstimationSample:
+        """
+        This helper method allows us to query additional samples for mixup & mosaic augmentations
+        that would be also passed through augmentation pipeline. Example:
+
+        ```
+          transforms:
+            - KeypointsBrightnessContrast:
+                brightness_range: [ 0.8, 1.2 ]
+                contrast_range: [ 0.8, 1.2 ]
+                prob: 0.5
+            - KeypointsHSV:
+                hgain: 20
+                sgain: 20
+                vgain: 20
+                prob: 0.5
+            - KeypointsLongestMaxSize:
+                max_height: ${dataset_params.image_size}
+                max_width: ${dataset_params.image_size}
+            - KeypointsMixup:
+                prob: ${dataset_params.mixup_prob}
+        ```
+
+        In the example above all samples in mixup will be forwarded through KeypointsBrightnessContrast, KeypointsHSV,
+        KeypointsLongestMaxSize and only then mixed up.
+
+        :param sample:         Input data sample
+        :param transforms:     List of transformations to apply
+        :param load_sample_fn: A method to load additional samples if needed
+        :param min_bbox_area:  Min bbox area of the pose instances to keep them
+        :return:               Transformed sample
+        """
+        applied_transforms_so_far = []
+        for t in transforms:
+            if t.additional_samples_count == 0:
+                sample = t.apply_to_sample(sample)
+                applied_transforms_so_far.append(t)
+            else:
+                additional_samples = [load_sample_fn() for _ in range(t.additional_samples_count)]
+                additional_samples = [
+                    cls._apply_transforms(
```

Comment:
In object detection we don't apply the transforms we did on the original sample (currently), but I actually like this.

Perhaps, if we want to preserve the behaviour and re-use this code, can we add a flag, something like "apply_transforms_on_additional_samples" which can be set to true ?

Also, if we are already applying these on the additinal samples, we can just load them in the beginning and it would perhaps be a bit more clear (this is just a suggestion, your call).

Reply:
In this case we can move Mixup/Mosaic to the top of transforms. I will update the docs to state this more clearly

Reply:
Fixed in c7971002071494f217d38258a9466266aa1ba38f